feat: OPTIC-478: Implement Feature Flag to Control Visibility and Access of Cloud Storage Connectors for Managers

yyassi-heartex opened the pull request:
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)



#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [x] Backend (API)
- [x] Frontend



### Describe the reason for change
adding a feature flag to toggle cloud storage view and access for managers



#### What does this fix?
_(if this is a bug fix)_



#### What is the new behavior?
_(if this is a breaking or feature change)_



#### What is the current behavior?
_(if this is a breaking or feature change)_



#### What libraries were added/updated?
_(list all with version changes)_



#### Does this change affect performance?
_(if so describe the impacts positive or negative)_



#### Does this change affect security?
_(if so describe the impacts positive or negative)_



#### What alternative approaches were there?
_(briefly list any if applicable)_



#### What feature flags were used to cover this change?
_(briefly list any if applicable)_



### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [ ] No
- [ ] Not sure (briefly explain the situation below)



### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [ ] integration
- [ ] unit



### Which logical domain(s) does this change affect?
_(for bug fixes/features, be as precise as possible. ex. Authentication, Annotation History, Review Stream etc.)_

yyassi-heartex committed:
feat: OPTIC-478: Implement Feature Flag to Control Visibility and Access of Cloud Storage Connectors for Managers

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8007008284

yyassi-heartex committed:
updating settings page to check new FF

yyassi-heartex committed:
Merge branch 'fb-optic-478/cloud-storage' of github.com:heartexlabs/label-studio into fb-optic-478/cloud-storage

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8007095678

yyassi-heartex committed:
fixing inversed logic

yyassi-heartex committed:
Merge branch 'fb-optic-478/cloud-storage' of github.com:heartexlabs/label-studio into fb-optic-478/cloud-storage

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8007674805

yyassi-heartex committed:
depending on FF we'll toggle perms

yyassi-heartex committed:
updating to a different capability

yyassi-heartex committed:
code cleanup

yyassi-heartex committed:
cleanup

wesleylima commented on label_studio/io_storages/all_api.py:
```diff
@@ -93,7 +94,10 @@ def get(self, request, **kwargs):
 
     parser_classes = (JSONParser, FormParser, MultiPartParser)
-    permission_required = all_permissions.projects_change
+    if flag_set('fflag_feat_all_optic_478_access_of_cloud_storage_connectors_short'):
```
This is evaluated at app load so it won't work with user/org ff targeting

makseq replied:
also there is no problem with this AllImportStorageListAPI::GET, we need to limit only POST/PATCH operations. I would say it might be dangerous to remove GET, probably we use storage info in some labeling processes for managers/reviewers/annotators.

makseq replied:
I don't think this class is the correct one at all.

yyassi-heartex committed:
switching to use licence flags

yyassi-heartex committed:
Merge branch 'develop' into fb-optic-478/cloud-storage

yyassi-heartex committed:
simplifying org license flag checks

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8022693721

yyassi-heartex committed:
renaming the file

yyassi-heartex committed:
Merge branch 'fb-optic-478/cloud-storage' of github.com:heartexlabs/label-studio into fb-optic-478/cloud-storage

yyassi-heartex committed:
fixing file rename

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8022970788

yyassi-heartex committed:
fixing var name

yyassi-heartex committed:
fixing lint

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8023278107

yyassi-heartex committed:
Merge branch 'develop' into fb-optic-478/cloud-storage

yyassi-heartex committed:
splitting license flags away from feature flags

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8051192967

yyassi-heartex committed:
cleaning up line endings

yyassi-heartex committed:
Merge branch 'fb-optic-478/cloud-storage' of github.com:heartexlabs/label-studio into fb-optic-478/cloud-storage

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8054091616

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
lint fix

yyassi-heartex committed:
lint clean up

yyassi-heartex committed:
Merge branch 'develop' into fb-optic-478/cloud-storage

makseq commented on label_studio/io_storages/all_api.py:
```diff
@@ -146,3 +151,8 @@ def list(self, request, *args, **kwargs):
         )
         return Response(list_responses)
+
+    def initial(self, request, *args, **kwargs):
```
I think we also need to add the similar check in PATCH request in ImportStorageDetailAPI and ExportStorageDetailAPI

bmartel committed:
use a permissions class instead of overriding all the various methods in the cbv

bmartel committed:
linting

bmartel committed:
update to point to overrideable storage permission class

bmartel committed:
fixing naming

bmartel committed:
wrong import

bmartel committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-478/cloud-storage

bmartel committed:
update storages to use permission class

bmartel committed:
unused func

bmartel committed:
Merge branch 'develop' into fb-optic-478/cloud-storage

nikitabelonogov commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8139264546)
> Successfully merged:  7 files changed, 28 insertions(+), 28 deletions(-)

nikitabelonogov committed:
Merge branch 'develop' into 'fb-optic-478/cloud-storage'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8139264546
